packages/core-data/src/queried-data/reducer.js

```diff
@@ -284,6 +285,148 @@ const queries = ( state = {}, action ) => {
+							value
+						);
+						return accumulator;
+					}, {} ),
+				},
+			};
+	}
+	return state;
+}
+
+const receiveRevisionQueries = compose( [
+	// Limit to matching action type so we don't attempt to replace action on
+	// an unhandled action.
+	ifMatchingAction( ( action ) => 'query' in action ),
+
+	// Inject query parts into action for use both in `onSubKey` and reducer.
+	replaceAction( ( action ) => {
+		// `ifMatchingAction` still passes on initialization, where state is
+		// undefined and a query is not assigned. Avoid attempting to parse
+		// parts. `onSubKey` will omit by lack of `stableKey`.
+		if ( action.query ) {
+			return {
+				...action,
+				...getQueryParts( action.query ),
+			};
+		}
+
+		return action;
+	} ),
+
+	onSubKey( 'context' ),
+
+	// Queries shape is shared, but keyed by query `stableKey` part. Original
+	// reducer tracks only a single query object.
+	onSubKey( 'stableKey' ),
+] )( ( state = {}, action ) => {
+	const { type, page, perPage, key = DEFAULT_ENTITY_KEY } = action;
+	if ( type !== 'RECEIVE_ITEM_REVISIONS' ) {
+		return state;
+	}
+
+	return {
+		itemIds: getMergedItemIds(
+			state?.itemIds || [],
+			action.items.map( ( item ) => item[ key ] ),
+			page,
+			perPage
+		),
+		meta: action.meta,
+	};
+} );
+
+export function revisionItemIsComplete( state = {}, action ) {
+	switch ( action.type ) {
+		case 'RECEIVE_ITEM_REVISIONS': {
+			const context = getContextFromAction( action );
+			const { query, key = DEFAULT_ENTITY_KEY } = action;
+			const queryParts = query ? getQueryParts( query ) : {};
+			const isCompleteQuery =
+				! query || ! Array.isArray( queryParts.fields );
+
+			return {
+				...state,
+				[ context ]: {
+					...state[ context ],
+					...action.items.reduce( ( result, item ) => {
+						const itemId = item[ key ];
+
+						// Defer to completeness if already assigned. Technically the
+						// data may be outdated if receiving items for a field subset.
+						result[ itemId ] =
+							state?.[ context ]?.[ itemId ] || isCompleteQuery;
+
+						return result;
+					}, {} ),
+				},
+			};
+		}
+	}
+
+	return state;
+}
+
+/**
+ * Reducer tracking queries state.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Next state.
+ */
+const revisionQueries = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case 'RECEIVE_ITEM_REVISIONS':
+			return receiveRevisionQueries( state, action );
+		default:
+			return state;
+	}
+};
+
+const revisionsReducer = combineReducers( {
+	items: revisionItems,
+	itemIsComplete: revisionItemIsComplete,
+	queries: revisionQueries,
+} );
+
+export const revisionsQueriedDataReducer = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case 'RECEIVE_ITEM_REVISIONS':
+			const { key: parentId } = parseEntityName( action.name );
+			return {
+				...state,
+				[ parentId ]: revisionsReducer( state[ parentId ], action ),
+			};
+		default:
+			return state;
+	}
+};
```

Comment:
I don't understand why you added these reducers to the queried-data reducer (but used them elsewhere). Can you clarify? In my mental modal, queried data is about having items and queries of a given object type (can be an entity and revision) and you use this reducer as a "helper" in other reducers. So I'm not sure I understand the additions here.

Reply:
It was to make it easier to create a separate entry in the redux store to process incoming items from the `RECEIVE_ITEM_REVISIONS` from `RECEIVE_ITEMS` actions.

I think you had a good alternative idea to update the action before calling the queriedReducer in the main reducer above, so all of this may not be necessary. I'll try that out, thanks!